app/celery/broadcast_message_tasks.py

```diff
@@ -27,12 +103,19 @@ def send_broadcast_event(broadcast_event_id):
         )
 
 
-@notify_celery.task(name="send-broadcast-provider-message")
+# max_retries=None: retry forever
+@notify_celery.task(bind=True, name="send-broadcast-provider-message", max_retries=None)
 @statsd(namespace="tasks")
-def send_broadcast_provider_message(broadcast_event_id, provider):
+def send_broadcast_provider_message(self, broadcast_event_id, provider):
     broadcast_event = dao_get_broadcast_event_by_id(broadcast_event_id)
 
-    broadcast_provider_message = create_broadcast_provider_message(broadcast_event, provider)
+    check_provider_message_should_send(broadcast_event, provider)
+
+    # the broadcast_provider_message will already exist if we retried previously
```

Comment:
Optional: Change word `will` to `may` as I'm not sure it's guaranteed it will exist?